src/plugins/irc-events/link.js

```diff
@@ -33,43 +31,45 @@ module.exports = function(client, chan, msg) {
 	// Remove all IRC formatting characters before searching for links
 	const cleanText = cleanIrcMessage(msg.text);
 
-	// We will only try to prefetch http(s) links
-	const links = findLinks(cleanText).filter((w) => linkRegex.test(w.link));
-
-	if (links.length === 0) {
-		return;
-	}
+	// Only preview the first 5 URLs in message to avoid abuse
+	let allowedLinks = 5;
+
+	msg.previews = findLinks(cleanText).reduce((cleanLinks, link) => {
+		const url = normalizeURL(link.link);
+
+		if (url !== null && allowedLinks-- > 0 && !cleanLinks.some((l) => l.link === link.link)) {
```

Comment:
Mind adding a comment for this check? It's one hell of a statement :D
I understand it checks that we haven't fetched 5 links yet + no duplicates, correct?

I think it would be even nicer if you were to split it a bit, like for example `allowedLinks-- > 0` is a bit weird to find in an `if`. No big deal though.

Also, I'm surprised there is no test for duplicate URLs (test that we don't fetch twice), do you think you could add a test for it, if that's something we can test?

Reply:
Yeah I'll split the if and add a test.

Reply:
Done.